src/java.xml/share/classes/com/sun/org/apache/xerces/internal/impl/xpath/regex/RegularExpression.java

```diff
@@ -2015,7 +2016,7 @@ static final class Context {
         int limit;
         int length;
         Match match;
-        boolean inuse = false;
+        private Semaphore inuse = new Semaphore(1, true);
```

Comment:
Does it require fairness? The existing code using object monitors so I assume not. I assume "inuse" can be final.